torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -1117,18 +1131,28 @@ void CudaCodeGen::call_raw(const std::vector<void*>& raw_args) {
       gpu_block_extents_v[i] = immediateAs<int64_t>(gpu_block_extents[i]);
       continue;
     }
+#ifdef TORCH_ENABLE_LLVM
+    gpu_block_extents_v[i] =
+        block_extents_compiled_[i].value<int64_t>(raw_args);
```

Comment:
Maybe we can avoid introducing `block_extents_compiled_` and instead pick `SimpleIREvaluator` or `LLVMCodegen` depending on `TORCH_ENABLE_LLVM`? Ie. we always try to pre-compile it in `Initialize`, but if there is no LLVM, we do that with IREval.

Reply:
We can precompile always, yes, thats a good idea. Will make that change.

But we will still need to introduce `block_extents_compiled_` and `thread_extents_compiled_`. I don't see how we can avoid that.